### ISSUE TITLE
Make reverse route helpers return a non-primitive string with method attribute

### ISSUE THREAD

**Issue by alexdavid** [alexdavid/exprestive#19]
@charlierudolph and I discussed #16 and came to the conclusion that while rails has magic around models, Exprestive should just be a routing library and therefore reverse routes need to be different for each action, and they should also return the method for form helpers to use.

This is the syntax we discussed:
- `paths.user(123)` would still return a string to the reverse route `/users/123`
- `paths.user(123).method` would return a string of the http method

This can be achieved with non-primitive strings:

``` coffeescript
str = new String '/users/123'
str.method = 'GET'
```

This way, as charlie mentioned, users of this library can easily implement a form helper:

``` jade
//- form_helper.jade
mixin form(path)
  form(action=path, method=path.method)
    if block 
      block

//- index.jade
+form(paths.users(123))
```

@jondistad @kevgo please let me know what you think

**Comment by charlierudolph** [comment]
Thinking about it more I think I would prefer just a simple hash (or Route object) with url and method attributes. Maybe have two helpers, one that returns a route and one that just returns the path.

``` coffeescript
res.locals.routes # functions return Route object
res.locals.paths  # functions return URLs as strings
```

``` jade
//- form_helper.jade
mixin form(route)
  form(action=route.url, method=route.method)
    if block 
      block

//- link_helper.jade
mixin link(text, path)
  a(href=path.url)
    text

//- index.jade
+link(paths.users())
+form(routes.updateUser(123))
```

**Comment by catharinejm** [comment]
I agree with @charlierudolph. Good to keep the paths as paths, but an abstraction over them which encapsulates the full action (path + method) would be useful.

**Comment by alexdavid** [comment]
I like this, what do you guys want to do for restful routes then?  I think both routes and paths should have the same name

**Comment by charlierudolph** [comment]
Please explain your last comment Alex. I'm unsure what you are referring to

**Comment by alexdavid** [comment]
Sorry, I'm talking about `update`, `create`, and `destroy` actions.

If we have separate routes object they can't be named the same as `show` and `index` because they have different methods.

Should paths contain a subset of the routes object since "user" and "update_user" would be the same path?

```
paths = { "user", "users", "new_user", "edit_user" }
routes = { "user", "users", "new_user", "edit_user", "update_user", "create_user", "destroy_user" }
```

**Comment by charlierudolph** [comment]
I would think paths and routes have the same keys but just different values. Thinking about routes added by something other then `resources` I don't think its makes sense to restrict what goes into paths

**Comment by kevgo** [comment]
I get the first path (separate paths and methods), but not the second part (paths vs routes). Can somebody please lay this out more simply, with examples?  The solution that the conversation ended with here seems redundant, where routes are just a superset of paths.

**Comment by alexdavid** [comment]
This is the current proposal

``` coffeescript
paths.editUser(123)          # "/users/123/edit"
routes.editUser(123).url     # "/users/123/edit"
routes.editUser(123).method  # "GET"
```

**Comment by kevgo** [comment]
I don't like the duplication between `paths.foo` and `routes.foo.url` Can we go with just the `routes` version for now, and then introduce a shortcut syntax for paths later as needed? We could also override the `toString` method on the object returned by `routes.foo`, so that it returns the url. This allows it to be used directly for the path in templates (which call `toString` on objects).

``` html
<form action="<%= routes.editUser(123) %>"
```

**Comment by alexdavid** [comment]
This was the original idea for this PR. The difference between using a non-primitive string and setting the `toString` method on an object is non primitive strings also set `valueOf`, `length`, and all `String` prototype methods such as `slice`, etc...

**Comment by kevgo** [comment]
Non-primitive strings sounds like the way to go to me then. `routes` seems a better namespace for that, since this comes from the routes file.

**Comment by alexdavid** [comment]
Merged in [c1338d83](https://github.com/alexdavid/exprestive/commit/c1338d83040e1749425549c4e8094d5c51a76e43)
